### ISSUE TITLE
sf.Series attribute error when 'broadcasting' a single value to an empty index

### ISSUE THREAD

**Issue by ForeverWintr** [InvestmentSystems/static-frame#87]
static_frame's `Series` constructor will broadcast a single value to fill the index it's provided, e.g.:

```Python
import static_frame as sf

sf.Series('T', index=[1, 2, 3])
<Series>
<Index>
1        T
2        T
3        T
<int64>  <<U1>
```

However, if the index is empty, the constructor fails with an attribute error:

```Python
sf.Series('T', index=[])
Traceback (most recent call last):
  Python Shell, prompt 9, line 1
  File "/static_frame/static_frame/core/series.py", line 269, in __init__
    self._index = Index(range(len(self.values)),
builtins.AttributeError: values
```

I think it would be more consistent to broadcast the value to the empty index, producing an empty series:

```Python
sf.Series('T', index=[])
<Series>
<Index>
<int64>  <float64>
```

**Comment by flexatone** [comment]
Many thanks, @ForeverWintr, for posting this issue.

I would interpret this as a case where the user has given arguments that are in internal conflict, and an Exception should be raised instead of returning an empty Series. 

Said another way, this suggests that you should not be permitted to give a value and an empty index, just as giving a list of length 3 for data and an list of length 2 for index raises an Exception. Returning an empty Series from such an initialization silently throws away the data value intended for that Series.  

I do think, however, the case should be explicitly handled and raise an informative Exception, rather than the currently raised AttributeError.

With these considerations, @ForeverWintr , would you still encourage returning an empty Series in this scenario?

**Comment by ForeverWintr** [comment]
I would, for similar reasons as why we allow a frame with columns and no index: Sometimes you're constructing a frame with an index you receive from elsewhere, and you want Static Frame to behave consistently. 

I don't think the data/index length mismatch argument applies here, because the broadcast behavior already violates it. If `sf.Series('T', index=[])` raises an exception, creating a broadcasted series will work for all possible index lengths except one, and will require a branch anywhere we're creating a broadcasted series from an index we don't control. 

In other words, I see `sf.Series('T', index=x)` as asking for an N ( `N = len(x)`) length series of 'T's. Why would 0 not be a valid N?

I think this behavior is analogous to a list comprehension in Python:

```python
['T' for x in index] # Will return an empty list if index is empty
```

Actually, using this list comprehension in the Series constructor results in the desired behavior by bypassing the broadcast:

```python
index = []
sf.Series(['T' for x in index], index=[x for x in index])
<Series>
<Index>
<int64>  <float64>
```

**Comment by flexatone** [comment]
Thanks, @ForeverWintr, for your further thoughts on this. While I think the comparison to a list comprehension is not exactly right (as that involves control flow, i.e., looping, which is quite different than instance initialization), I am compelled by your point that the broadcasting behavior, as a special case, should be valid for all of N, including zero. Supporting the implementation of this also aligned with some other enhancements to `Series` initialization.

This functionality will be available in the 0.4.0 release.
